Cython/Coverage.py

```diff
@@ -35,11 +35,11 @@ def _find_c_source(base_path):
 
 def _find_dep_file_path(main_file, file_path):
     abs_path = os.path.abspath(file_path)
-    if file_path.endswith('.pxi') and not os.path.exists(abs_path):
-        # include files are looked up relative to the main source file
-        pxi_file_path = os.path.join(os.path.dirname(main_file), file_path)
-        if os.path.exists(pxi_file_path):
-            abs_path = os.path.abspath(pxi_file_path)
+    if not os.path.exists(abs_path):
+        # files are looked up relative to the main source file
```

Comment:
This is actually not true. In the compiler, it really only applies to `include` files. That does not make this change feel like the right fix in general. It might still be the right thing to do for the source lookup relative to a .c/.cpp/etc. file, i.e. one of the two cases (in `_read_source_lines()`) where this function is used. Not sure about the other call.

Reply:
OK, I have just pushed a small change to explicitly enable the search of files relative to the main source. Relative search is now enabled only in calls made from `_read_source_lines()`.

Honestly I don't have clear in mind all possible corner cases so please let me know if the proposed solution makes sense for you or if you have a better idea.